collect_app/src/test/java/org/odk/collect/android/http/OkHttpCaseInsensitiveHeadersTest.java

```diff
@@ -0,0 +1,83 @@
+package org.odk.collect.android.http;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Assert;
+import org.odk.collect.android.http.openrosa.okhttp.OkHttpCaseInsensitiveHeaders;
+
+import java.util.Set;
+import java.util.TreeSet;
+
+import okhttp3.Headers;
+
+public class OkHttpCaseInsensitiveHeadersTest {
```

Comment:
I was initially interested in if this would be a nice case for deploying a "contract testing" style like we use for `OpenRosaHttpInterface`. However it feel it might not be right as the "empty" implementation and this really only share one "contract" - if they are empty they return false. Interested if you have any thoughts on this however?

Reply:
I'm a bottom-up kind of dev. I'd rather have functional tests first, and I loathe Mock testing except as a last resort. Contract-based testing seems best suited to the boundary of external services or APIs, but internal interfaces should just be simple tests that rely on no shared state, can run in isolation and potentially even run in parallel.

Reply:
Right. With our OpenRosaHttpInterface we are drawing a boundary between our "App" and how it communicates with OpenRosa and I think you're right that for this case there isn't really that boundary.